src/main/kotlin/build/buf/gradle/ProtobufGradlePluginSupport.kt

```diff
@@ -75,18 +80,68 @@ internal fun Project.configureWriteWorkspaceYaml() {
 }
 
 abstract class WriteWorkspaceYamlTask : DefaultTask() {
+    private val yamlMapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
+
     @TaskAction
     fun writeWorkspaceYaml() {
-        val bufWork =
-            """
-                |version: v1
-                |directories:
-                ${workspaceSymLinkEntries()}
-            """.trimMargin()
+        val bufYaml = rewriteBufYaml(project.bufConfigFile())
+        logger.info("Writing generated buf.yaml:\n${yamlMapper.writeValueAsString(bufYaml)}")
+        yamlMapper.writeValue(File(bufbuildDir, "buf.yaml"), bufYaml)
+    }
 
-        logger.info("Writing generated buf.work.yaml:\n$bufWork")
+    private fun rewriteBufYaml(bufYamlFile: File?): MutableMap<String, Any> {
```

Comment:
Can we isolate this in its own class? I'd also like to get some unit tests in for this since it seems like it can be expressed as a pure function in isolation from integration tests.